portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -66,22 +67,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   }
 
   /** Fetches markers from Datastore. */
-  private List<Marker> getMarkers() {
+  private List<Marker> getMarkers(HttpServletRequest request){
     List<Marker> markers = new ArrayList<>();
     Query query = new Query(ENTITY_TITLE);
     PreparedQuery results = datastore.prepare(query);
+    Location location = getUserLocation(request);
 
-    for (Entity entity : results.asIterable()) {
-      double lat = (double) entity.getProperty("lat");
-      double lng = (double) entity.getProperty("lng");
-      String crime = (String) entity.getProperty("crimeType");
-      String date = (String) entity.getProperty("date");
-      String time = (String) entity.getProperty("time");
-      String address = (String) entity.getProperty("address");
-      String description = (String) entity.getProperty("description");
-
-      Marker marker = new Marker(lat, lng, crime, date, time, address, description);
-      markers.add(marker);
+    for(Entity entity: results.asIterable()){
+        double lat = (double) entity.getProperty("lat");
+        double lng = (double) entity.getProperty("lng");
+        String crime = (String) entity.getProperty("crimeType");
+        String date = (String) entity.getProperty("date");
+        String time = (String) entity.getProperty("time");
+        String address = (String) entity.getProperty("address");
+        String description = (String) entity.getProperty("description");
+        //fetches markers olny if they are inside the wanted area
+        if(haversineDistance(location, lat, lng) <= metersInAMile){ 
```

Comment:
You can delete your method `haversineDistance` since it doesn't do anything except call another method. Just call `haversinMeters(...)` directly here